test/integ/user_comments_integ_test.js

```diff
@@ -0,0 +1,278 @@
+            const commentsBody = commentsResult.text;
+            assert.match(commentsBody, new RegExp(new Date(postedComment.postedDate).toUTCString(), "g"));
+        });
+
+        it("should return removed image link within comment HTML if referenced image has been deleted", async () => {
+            // Upload an image
+            const uploadedImages = await addImagesForUser([
+                {
+                    fileName: "Black_tea_pot_cropped.jpg",
+                    mimeType: "image/jpeg",
+                },
+            ], TEST_USER);
+            // Write comment on image
+            const writeResult = await writeComment(agent, TEST_USER, uploadedImages[0].id, COMMENT_TEXT);
+            if (writeResult.statusCode !== 200) {
+                assert.fail(`Could not write comment, status code: ${writeResult.statusCode}, resp: ${JSON.stringify(writeResult.body)}`);
+            }
+            // Delete image
+            const deleteResult = await deleteImage(agent, uploadedImages[0].id);
+            if (deleteResult.statusCode !== 200) {
+                assert.fail(`Could not delete image, status code: ${deleteResult.statusCode}, resp: ${JSON.stringify(deleteResult.body)}`);
+            }
+            // Verify that 'removed' image link exists in API response
+            const commentsResult = await getUserComments(agent, TEST_USER, "html");
+            assert.equal(commentsResult.statusCode, 200);
+            const commentsBody = commentsResult.text;
+            assert.match(commentsBody, new RegExp(`"/images/${uploadedImages[0].id}"`, "g"));
+            assert.match(commentsBody, new RegExp(`"/images/removed.png"`, "g"));
+        });
+
+        it("should fail to return user comments if fail to retrieve image entry attributes from database", async () => {
+            // Upload an image
+            const uploadedImages = await addImagesForUser([
+                {
+                    fileName: "Black_tea_pot_cropped.jpg",
+                    mimeType: "image/jpeg",
+                },
+            ], TEST_USER);
+            // Write comment on image
+            const writeResult = await writeComment(agent, TEST_USER, uploadedImages[0].id, COMMENT_TEXT);
+            if (writeResult.statusCode !== 200) {
+                assert.fail(`Could not write comment, status code: ${writeResult.statusCode}, resp: ${JSON.stringify(writeResult.body)}`);
+            }
+            // Stub databaseOps findImageAttributes
+            const findImageAttributesStub = stub(databaseOps, "findImageAttributes").callsArgWith(1, new Error("Error finding image"), null);
+            // Verify that comments could not load due to an error
+            const commentsResult = await getUserComments(agent, TEST_USER, "html");
+            assert.equal(commentsResult.statusCode, 500);
+            const commentsBody = commentsResult.text;
+            const validateReport = htmlValidate.validateString(commentsBody);
```

Comment:
Consider making a helper function to validate HTML, maybe even set it up as a helper object that requires an html-validate object to setup.

Reply:
Will take care of this when I clean and consolidate the testing helper functions.